matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java

```diff
@@ -1127,106 +1151,156 @@ public boolean sendReadReceipt(final ApiCallback<Void> aRespCallback) {
+            sendReadMarkers(readMarkerEventId, summary.getReadReceiptEventId(), null);
+        }
+    }
+
+    /**
+     * Send a read receipt to the latest known event
+     */
+    public void sendReadReceipt() {
+        markAllAsRead(false, null);
+    }
+
+    /**
+     * Send the read receipt to the latest room message id.
+     *
+     * @param event send a read receipt to a provided event
+     * @param aRespCallback asynchronous response callback
+     * @return true if the read receipt has been sent, false otherwise
+     */
+    public boolean sendReadReceipt(Event event, final ApiCallback<Void> aRespCallback) {
+        return sendReadMarkers(null, (null != event) ? event.eventId : null, aRespCallback);
+    }
+
+    /**
+     * Forget the current read marker
+     * This will update the read marker to match the read receipt
+     *
+     * @param callback
+     */
+    public void forgetReadMarker(final ApiCallback<Void> callback) {
+        final RoomSummary summary = mStore.getSummary(getRoomId());
+        final String currentReadReceipt = summary.getReadReceiptEventId();
+        summary.setReadMarkerEventId(currentReadReceipt);
+        mStore.flushSummary(summary);
+        setReadMarkers(currentReadReceipt, currentReadReceipt, callback);
+    }
+
+    /**
+     * Send the read markers
+     *
+     * @param aReadMarkerEventId the new read marker event id (if null use the latest known event id)
+     * @param aReadReceiptEventId the new read receipt event id (if null use the latest known event id)
      * @param aRespCallback asynchronous response callback
-     * @return true if the read receipt request is sent, false otherwise
+     * @return true if the request is sent, false otherwise
      */
-    public boolean sendReadReceipt(Event anEvent, final ApiCallback<Void> aRespCallback) {
+    public boolean sendReadMarkers(final String aReadMarkerEventId, final String aReadReceiptEventId, final ApiCallback<Void> aRespCallback) {
         final Event lastEvent = mStore.getLatestEvent(getRoomId());
-        final Event fEvent;
 
         // reported by GA
         if (null == lastEvent) {
-            Log.e(LOG_TAG, "## sendReadReceipt(): no last event");
+            Log.e(LOG_TAG, "## sendReadMarkers(): no last event");
             return false;
         }
 
-        // the event is provided
-        if (null != anEvent) {
-            Log.d(LOG_TAG, "## sendReadReceipt(): roomId=" + getRoomId() + " to " + anEvent.eventId);
+        boolean hasUpdate = false;
 
-            // test if the message has already be read
-            if (getDataHandler().getStore().isEventRead(getRoomId(), getDataHandler().getUserId(), anEvent.eventId)) {
-                Log.d(LOG_TAG, "## sendReadReceipt(): the message was already read");
-                return false;
+        String readMarkerEventId = aReadMarkerEventId;
+        if (!TextUtils.isEmpty(aReadMarkerEventId)) {
+            if (aReadMarkerEventId.startsWith(getRoomId())) {
```

Comment:
MXSession.PATTERN_CONTAIN_MATRIX_MESSAGE_IDENTIFIER should be used.
I assume it is used to check if it is not a tmp event.